R-package/tests/testthat/test_Predictor.R

```diff
@@ -0,0 +1,17 @@
+context("Predictor")
+
+test_that("predictions do not fail for integer input", {
+    X <- as.matrix(as.integer(iris[, "Species"]), ncol = 1L)
+    y <- iris[["Sepal.Length"]]
+    dtrain <- lgb.Dataset(X, label = y)
+    fit <- lgb.train(
+        data = dtrain
+        , objective = "regression"
+        , verbose = -1L
+    )
+    X_double <- X_integer <- X[c(1L, 51L, 101L), , drop = FALSE]
```

Comment:
do you mind breaking this up into two lines please? We don't use double-`<-` anywhere else in the codebase, as I think it's a little difficult to read.

```r
X_double <- X[c(1L, 51L, 101L), , drop = FALSE]
X_integer <- X_double
```